core/primitives-core/src/serialize.rs

```diff
@@ -2,6 +2,13 @@ pub fn to_base58<T: AsRef<[u8]>>(input: T) -> String {
     bs58::encode(input).into_string()
 }
 
+/// Deprecated.  If you want to decode a CryptoHash, use CryptoHash::from_str.
+/// For anything else, don’t use base58.  This is still here because of
+/// deprecated RPC query format in RpcQueryRequest::parse.
+pub fn from_base58(s: &str) -> Result<Vec<u8>, Box<dyn std::error::Error + Send + Sync>> {
```

Comment:
In a follow-up, lets move this closed to the use-site, so that we don't have to mark it as deprecated

Reply:
Counter argument though is that it would add direct `bs58` dependency in `chain/jsonrpc` which we’ll forget to remove once `from_base58` is removed.